### PR TITLE
Update egui to version 0.29.1

### PR DESCRIPTION
**Pull request by BlueCookieFrog**
Updated the version of egui to the version 0.29.1.
I also updated the deprecated code.

Should make new things possible and also fix this issue #146

### CODE REVIEW

**Comment by BlueCookieFrog** [comment]
I've also made a couple of changes to the UI, but they would require this change to be merged

**Comment by trumank** [comment]
The diff is quite hard to follow so opted to run the update myself: 2a0e24587eefa4218e095548a75afc96ba7796f7
